sklearn/ensemble/weight_boosting.py

```diff
@@ -234,11 +234,47 @@ def staged_score(self, X, y, sample_weight=None):
             else:
                 yield r2_score(y, y_pred, sample_weight=sample_weight)
 
-    @property
-    def feature_importances_(self):
+    @staticmethod
+    def _coef_to_feature_importances_(coef, norm_type):
+        """Returns feature_importances_ by normalising coef
+
+        Parameters
+        ----------
+        coef : ndarray
+
+        norm_type: str
+            The kind of norm that should be used make data non-negative.
+            Accepted values: "l1" (absolute values), "l2" (squared values)
+
+        Returns
+        -------
+        feature_importances_ : array
+        """
+
+        if norm_type == "l1":
+            non_negative_coefs_ = abs(coef)
+        elif norm_type == "l2":
+            non_negative_coefs_ = coef ** 2
+        else:
+            raise ValueError(
+                "Invalid norm_type supplied for converting"
+                "coef_ to feature_importances_. Use either"
+                "`l1` or `l2` norm"
+            )
+        norm = non_negative_coefs_.sum()
+
+        return non_negative_coefs_ / norm
```

Comment:
If the problem is multiclass, this return value will still, as far as I can tell, be 2-dimensional. `feature_importances_` must be 1d. I had intended that the norm would be used to aggregate coefficients across classes, as is done in [`_get_feature_importances`](https://github.com/scikit-learn/scikit-learn/blob/5fd9e03cf3a2e58d3b76dfaf0692437a4d6e1eec/sklearn/feature_selection/from_model.py#L15).

Reply:
Alright, I will reuse `_get_feature_importances` function from `feature_selection` to extend `feature_importances_` attribute of AdaBoost base class. Thanks for the link.